compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1318,6 +1318,22 @@ private boolean validateErrorVariable(BLangErrorVariable errorVariable) {
             Map<String, BField> fieldMap = recordType.fields.stream()
                     .collect(Collectors.toMap(f -> f.name.value, f -> f));
 
+            if (isReasonSpecified(errorVariable)
```

Comment:
We can move this to L1317 right? The variables defined are not used in this block?